docs/queries/geo_polygon.rst

```diff
@@ -0,0 +1,15 @@
+Geo-Polygon Query
+==========================
+
+A query that matches documents with geo point within a defined set of polygons.
```

Comment:
@aprudhomme should we add the restrictions mentioned in the notes in https://lucene.apache.org/core/8_4_0/core/org/apache/lucene/geo/Polygon.html ?

Reply:
The important ones are in the `Polygon` comments, but I can add them here too.

I tried to do some digging, and it looks like some of the requirements have to do with conforming to the GeoJson spec. https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.6 Oddly, the winding order in the spec looks like the opposite of the class notes. The spec also says it is ok to ignore the wrong winding order.

I also see that the lucene class detects the winding order in the constructor, so it really might not be vital to functionality. I think I will try writing some tests that use the opposite order, and relax the requirements if they work.